spi/src/main/java/org/opensearch/security/spi/resources/client/NoopResourceSharingClient.java

```diff
@@ -0,0 +1,118 @@
+
+/**
+ * Creates a noop client that will be used when security is disabled,
+ *
+ * @opensearch.experimental
+ */
+public final class NoopResourceSharingClient implements ResourceSharingClient {
+
+    private static final Logger log = LogManager.getLogger(NoopResourceSharingClient.class);
+
+    /**
+     * Constructs a NoopResourceSharingClient
+     */
+    public NoopResourceSharingClient() {}
+
+    /**
+     * Returns true as a no-op implementation.
+     *
+     * @param resourceId    The ID of the resource to verify.
+     * @param resourceIndex The index in which the resource resides.
+     * @param listener      Callback that receives {@code true} if access is granted, {@code false} otherwise.
+     */
+    @Override
+    public void verifyResourceAccess(String resourceId, String resourceIndex, ActionListener<Boolean> listener) {
+        String message = getFeatureDisabledReason() + " " + "Access to resource is automatically granted.";
+        log.debug("{}", message);
+        listener.onResponse(true);
+    }
+
+    /**
+     * Throws 501 exception as a no-op implementation.
+     *
+     * @param resourceId    The ID of the resource to share.
+     * @param resourceIndex The index containing the resource.
+     * @param recipients    The recipients of the resource, including users, roles, and backend roles.
+     * @param listener      Callback receiving the updated {@link ResourceSharing} document.
+     */
+    @Override
+    public void shareResource(
+        String resourceId,
+        String resourceIndex,
+        SharedWithActionGroup.ActionGroupRecipients recipients,
+        ActionListener<ResourceSharing> listener
+    ) {
+        handleSecurityDisabled("Resource is not shareable.", listener);
+    }
+
+    /**
+     * Throws 501 exception as a no-op implementation.
+     *
+     * @param resourceId        The ID of the resource.
+     * @param resourceIndex     The index containing the resource.
+     * @param entitiesToRevoke  A map of entities whose access is to be revoked.
+     * @param listener          Callback receiving the updated {@link ResourceSharing} document.
+     */
+    @Override
+    public void revokeResourceAccess(
+        String resourceId,
+        String resourceIndex,
+        SharedWithActionGroup.ActionGroupRecipients entitiesToRevoke,
+        ActionListener<ResourceSharing> listener
+    ) {
+        handleSecurityDisabled("Resource access is not revoked.", listener);
+    }
+
+    /**
+     * Throws 501 exception as a no-op implementation.
+     *
+     * @param resourceIndex The index to search for accessible resources.
+     * @param listener      Callback receiving a set of {@link ShareableResource} instances.
+     */
+    @Override
+    public <T extends ShareableResource> void listAllAccessibleResources(String resourceIndex, ActionListener<Set<T>> listener) {
+        handleSecurityDisabled("Unable to list all accessible resources.", listener);
+    }
```

Comment:
Do we want to just return all instead of not returning anything? Similar to verifyResourceAccess method?

Reply:
I feel like this should be handled by plugin, but I'm open to suggestions. Remember, if we decide to implement it here, we will need a parser which can only be passed when registering itself as resource-sharing plugin. We store these inside ResourcePluginInfo in sec plugin, but in No-op case we would need that information to be passed inside this API.